lib/src/Uncertainty/Process/DiscreteMarkovChain.cxx

```diff
@@ -294,6 +294,22 @@ void DiscreteMarkovChain::setMesh(const Mesh & mesh)
   ProcessImplementation::setMesh(mesh);
 }
 
+/* DOT export */
+void DiscreteMarkovChain::exportToDOTFile(const FileName & filename) const
+{
+  std::ofstream dotFile(filename.c_str());
+  if (dotFile.fail())
+    throw FileOpenException(HERE) << "Could not open file " << filename;
+  dotFile.imbue(std::locale("C"));
+  dotFile << "digraph " << getName() << " {\n";
+  for (UnsignedInteger j = 0; j < transitionMatrix_.getNbRows(); ++ j)
+    for (UnsignedInteger i = 0; i < transitionMatrix_.getNbRows(); ++ i)
+      if (transitionMatrix_(i, j) > 0.0)
+        dotFile << j+1 << "->" << i+1 << " [label=" << transitionMatrix_(i, j) << "]\n";
```

Comment:
It should be `dotFile << j << "->" << i << " [label=" << transitionMatrix_(i, j) << "]\n";` as the state space is `[[0,...,n-1]]` and not `[[1,...,n]]`.
What's about using some OT-ish colors? Something like:
```
  dotFile << "layout=circo\n";
  dotFile << "resolution=300\n";
  dotFile << "node[shape=circle, color=red]";
  for (UnsignedInteger j = 0; j < transitionMatrix_.getNbRows(); ++ j)
    for (UnsignedInteger i = 0; i < transitionMatrix_.getNbRows(); ++ i)
      if (transitionMatrix_(i, j) > 0.0)
        dotFile << j+1 << "->" << i+1 << " [label=" << transitionMatrix_(i, j) <<  ", weight=0.5, fontsize=12, color=green]\n";
```

Reply:
Anyway, good job!

Reply:
it's a bit too much for my eyes : )
![markov2](https://user-images.githubusercontent.com/3832365/59145917-c0b5c000-89ea-11e9-8a65-4d930b6ca7d4.png)

Reply:
But I added resourcemap keys to customize the graph settings

Reply:
You don't like pizzas ;-)?